.github/workflows/reusable-build.yml

```diff
@@ -188,98 +194,123 @@ jobs:
+      #     registry: ghcr.io
+      #     username: ${{ github.actor }}
+      #     password: ${{ secrets.GITHUB_TOKEN }}
 
-      # Sign container
-      - uses: sigstore/cosign-installer@v3.4.0
-        if: github.event_name != 'pull_request'
+      # # Sign container
+      # - uses: sigstore/cosign-installer@v3.4.0
+      #   if: github.event_name != 'pull_request'
 
-      - name: Sign container image
-        if: github.event_name != 'pull_request'
-        run: |
-          cosign sign -y --key env://COSIGN_PRIVATE_KEY ${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}@${TAGS}
-        env:
-          TAGS: ${{ steps.push.outputs.digest }}
-          COSIGN_EXPERIMENTAL: false
-          COSIGN_PRIVATE_KEY: ${{ secrets.SIGNING_SECRET }}
+      # - name: Sign container image
+      #   if: github.event_name != 'pull_request'
+      #   run: |
+      #     cosign sign -y --key env://COSIGN_PRIVATE_KEY ${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}@${TAGS}
+      #   env:
+      #     TAGS: ${{ steps.push.outputs.digest }}
+      #     COSIGN_EXPERIMENTAL: false
+      #     COSIGN_PRIVATE_KEY: ${{ secrets.SIGNING_SECRET }}
 
-      - name: Echo outputs
-        if: github.event_name != 'pull_request'
+      - name: Output single image
+        id: generate-outputs
+        env:
+          # DIGEST: ${{ steps.push.outputs.digest }}
+          DIGEST: ${{ matrix.image_flavor == 'main' && 'sha256:21e6bb281ebaa598f05dc23e74b704a8fd67754f29c224e246091001670cf61' || 'sha256:5d2383447773904988d4d97d695ceee3ff43484b1015554a8394789a1be2e8c8' }}
+          IMAGE_REGISTRY: ${{ env.IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}
         run: |
-          echo "${{ toJSON(steps.push.outputs) }}"
+          echo "image=${IMAGE_REGISTRY}:${DIGEST}" >> $GITHUB_OUTPUT
 
   check:
     name: Check all ${{ inputs.brand_name }} ${{ inputs.fedora_version }} builds successful
-    if: ${{ !cancelled() }}
+    if: always()
     runs-on: ubuntu-latest
     needs: [build_container]
+    outputs:
+      images: ${{ steps.generate-outputs.outputs.images }}
     steps:
-      - name: Exit on failure
-        if: ${{ needs.build_container.result == 'failure' }}
-        shell: bash
-        run: exit 1
-      - name: Exit
-        shell: bash
-        run: exit 0
+      - name: Check Jobs
+        env:
+          JOBS: ${{ toJson(needs) }}
+        run: |
+          echo "Job status:"
+          echo $JOBS | jq -r 'to_entries[] | " - \(.key): \(.value.result)"'
+
+          for i in $(echo $JOBS | jq -r 'to_entries[] | .value.result'); do
+            if [ "$i" != "success" ] && [ "$i" != "skipped" ]; then
+              echo ""
+              echo "Status check not okay!"
+              exit 1
+            fi
+          done
+
+      - name: Create output
+        id: generate-outputs
+        env:
+          JOBS: ${{ toJson(needs) }}
+        run: |
+          images=$(echo $JOBS | jq -c '[to_entries[] | .value.outputs.image_full]')
```

Comment:
Current problem is the `toJson(needs)` does not contain information about all matrix jobs.  Only the one that finished most recently.

Will change each job to upload to a common artifact, and extract that in the final outputs step.  It's hacky, but should work.

Reply:
Of course that didn't work.  GitHub changed the way the upload-artifact action works in the recent v4 release.

Next idea is to create a job before the container builds which generates the matrix and sets the outputs. So set the output before fanning out to each individual build.